### ISSUE TITLE
Compile failure w/ base-4.6

### ISSUE THREAD

**Issue by hvr** [well-typed/generics-sop#41]
I went ahead and revised in a tighter lower bound on `base` for the affected release (https://hackage.haskell.org/package/generics-sop-0.3.0.0/revisions/)

```
Configuring component lib from generics-sop-0.3.0.0...
Preprocessing library generics-sop-0.3.0.0...
[ 1 of 14] Compiling Generics.SOP.Sing ( src/Generics/SOP/Sing.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/Sing.o )
[ 2 of 14] Compiling Generics.SOP.Constraint ( src/Generics/SOP/Constraint.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/Constraint.o )
[ 3 of 14] Compiling Generics.SOP.BasicFunctors ( src/Generics/SOP/BasicFunctors.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/BasicFunctors.o )
[ 4 of 14] Compiling Generics.SOP.Classes ( src/Generics/SOP/Classes.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/Classes.o )
[ 5 of 14] Compiling Generics.SOP.NP  ( src/Generics/SOP/NP.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/NP.o )
[ 6 of 14] Compiling Generics.SOP.Metadata ( src/Generics/SOP/Metadata.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/Metadata.o )
[ 7 of 14] Compiling Generics.SOP.Type.Metadata ( src/Generics/SOP/Type/Metadata.hs, /tmp/matrix-worker/1493460027/dist-newstyle/build/x86_64-linux/ghc-7.6.3/generics-sop-0.3.0.0/build/Generics/SOP/Type/Metadata.o )

src/Generics/SOP/Type/Metadata.hs:144:7:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:144:22:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:148:8: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:149:8: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:153:7:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:153:22:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:157:8: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:158:8: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:196:11:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:197:44: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:200:7:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:200:45:
    Not in scope: type constructor or class `KnownNat'

src/Generics/SOP/Type/Metadata.hs:204:8: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:206:21: Not in scope: `natVal'

src/Generics/SOP/Type/Metadata.hs:208:11:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:210:15: Not in scope: `symbolVal'

src/Generics/SOP/Type/Metadata.hs:246:10:
    Not in scope: type constructor or class `KnownSymbol'

src/Generics/SOP/Type/Metadata.hs:247:36: Not in scope: `symbolVal'
```

**Comment by hvr** [comment]
It turns out, this would have been caught by Travis CI as well (see https://travis-ci.org/well-typed/generics-sop/jobs/227103927 ), however GHC 7.6.3 has been marked as allowed to fail in the configuration...

**Comment by kosmikus** [comment]
I know, sorry. This was just an oversight from me. Travis had caught it, and I had marked it as allowed failure in reaction, because at the point it wasn't 100% clear whether the final version might restore compatibility with 7.6 or not. I should have bumped the lower bound prior to release though. Thanks for fixing it on Hackage. I've also committed the change to master as f60c0d8a3bbf73d60864aeb2fa9c7e0e1895fde1.
